# HG changeset patch
# User Michael Goffioul <michael.goffioul@gmail.com>
# Date 1204658176 18000
#      Tue Mar 04 14:16:16 2008 -0500
# Node ID 9cbf1e2011a3db20aa66b159c7ab3d92c10a4c83
# Parent  40ca9fb9a3143ebd1f0b822cd28547f269b24c1a
pkg.m: adapt to changes in isspace for cell arrays of strings.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2008-03-04  Michael Goffioul <michael.goffioul@gmail.com>
+
+	* pkg/pkg.m (pkg:configure_make): Make it work with recent changes in
+	isspace handling with cell arrays of strings.
+
 2008-03-04  Ben Abbott <bpabbott@mac.com>
 
 	* polynomial/polyfit.m: Modified tests to respect a relative tolerance.
 
 2008-02-29  John W. Eaton  <jwe@octave.org>
 
 	* plot/print.m: Handle gif and jpg devices.
 
diff --git a/scripts/pkg/pkg.m b/scripts/pkg/pkg.m
--- a/scripts/pkg/pkg.m
+++ b/scripts/pkg/pkg.m
@@ -1291,33 +1291,33 @@ function configure_make (desc, packdir, 
       idx = [];
     else
       idx = cellfun (@(x) is_architecture_dependent (x), filenames);
     endif
     archdependent = filenames (idx);
     archindependent = filenames (!idx);
 
     ## Copy the files
-    if (! all (isspace (filenames)))
+    if (! all (isspace ([filenames{:}])))
 	if (! exist (instdir, "dir")) 
 	  mkdir (instdir);
 	endif
-	if (! all (isspace (archindependent)))
+	if (! all (isspace ([archindependent{:}])))
 	  if (verbose)
 	    printf ("copyfile");
 	    printf (" %s", archindependent{:});
 	    printf ("%s\n", instdir);
 	  endif
 	  [status, output] = copyfile (archindependent, instdir);
 	  if (status != 1)
 	    rm_rf (desc.dir);
 	    error ("Couldn't copy files from 'src' to 'inst': %s", output);
 	  endif
         endif
-	if (! all (isspace (archdependent)))
+	if (! all (isspace ([archdependent{:}])))
 	  if (verbose)
 	    printf ("copyfile");
 	    printf (" %s", archdependent{:});
 	    printf (" %s\n", archdir);
 	  endif
 	  if (! exist (archdir, "dir")) 
 	    mkdir (archdir);
 	  endif
